### ISSUE TITLE
Duplicated creation of RoleBindings with 0.8.2

### ISSUE THREAD

**Issue by msiuts** [FairwindsOps/rbac-manager#77]
As a follow up to https://github.com/FairwindsOps/rbac-manager/issues/74 we now see that the new version of rbac-manager seem to try to add some ClusterRoleBinding and RoleBindings twice, which leads to an error in the logs. This happens when we add a user for exmaple.

From looking at the code, I suspect that these roles are existing twice in the `parsedClusterRoleBindings` but I did not dig deep enough to be sure.

This does not happen when there is only exactly one ClusterRolebinding is defined, but only when there are multiple.

Here are the logs: 

```
time="2019-09-13T06:57:06Z" level=info msg="Reconciling RBACDefinition human-access"
time="2019-09-13T06:57:06Z" level=info msg="Deleting Cluster Role Binding: XXX"
time="2019-09-13T06:57:06Z" level=info msg="Deleting Cluster Role Binding: YYY"
time="2019-09-13T06:57:06Z" level=info msg="Creating Cluster Role Binding: YYY"
time="2019-09-13T06:57:06Z" level=info msg="Creating Cluster Role Binding: YYY"
time="2019-09-13T06:57:06Z" level=info msg="Creating Cluster Role Binding: XXX"
time="2019-09-13T06:57:06Z" level=error msg="Error creating Cluster Role Binding: clusterrolebindings.rbac.authorization.k8s.io \"YYY\" already exists"
time="2019-09-13T06:57:06Z" level=info msg="Creating Cluster Role Binding: XXX"
time="2019-09-13T06:57:06Z" level=error msg="Error creating Cluster Role Binding: clusterrolebindings.rbac.authorization.k8s.io \"XXX\" already exists"
time="2019-09-13T06:57:06Z" level=info msg="Deleting Role Binding ZZZ"
time="2019-09-13T06:57:07Z" level=info msg="Deleting Role Binding AAA"
time="2019-09-13T06:57:07Z" level=info msg="Creating Role Binding: ZZZ"
time="2019-09-13T06:57:07Z" level=info msg="Creating Role Binding: ZZZ"
time="2019-09-13T06:57:07Z" level=info msg="Creating Role Binding: AAA"
time="2019-09-13T06:57:07Z" level=error msg="Error creating Role Binding: rolebindings.rbac.authorization.k8s.io \"ZZZ\" already exists"
time="2019-09-13T06:57:07Z" level=info msg="Creating Role Binding: AAA"
time="2019-09-13T06:57:07Z" level=error msg="Error creating Role Binding: rolebindings.rbac.authorization.k8s.io \"AAA\" already exists"
```

**Comment by msiuts** [comment]
After adding some logging I think this is a race condition: The Delete of the ClusterRoleBinding is triggering another run of the reconciling.

**Comment by msiuts** [comment]
Hi @sudermanjr would it be possible to take look at this issue / the pr? This kind of blocks me from upgrading at the moment and I would also be able to do some changes on the fix if the mutex is not the preferred way of achieving this.

**Comment by sudermanjr** [comment]
I'll take a look today. Sorry about the delay

**Comment by sudermanjr** [comment]
Fixed by #78
